line_profiler/ipython_extension.py

```diff
@@ -0,0 +1,150 @@
+from io import StringIO
+
+from .line_profiler import LineProfiler
+
+LineProfilerMagics = None
+
+
+def load_ipython_extension(ip):
+    """ API for IPython to recognize this module as an IPython extension.
+    """
+
+    # Import IPython inside the function to load IPython only if necessary
```

Comment:
I definitely don't want to define this class as a local object. That can cause weird corner cases on win32 where objects need to be defined in the global scope for them to be pickleable, and hence shared across a multiprocessing queue. An easy workaround is to make an `_ipython_extension_backend.py` module (If the `ipython_extension.py` module can't serve that purpose) that defines the classes of interest, but is only conditionally imported.

Reply:
I see what you mean, I'll try something like this.